### PR TITLE
Small doc fixes

### PR DESCRIPTION
**Pull request by hendi**
plus a few typos

### CODE REVIEW

File: Guide/mail.markdown [hunk 1]
```diff
@@ -196,4 +221,62 @@ instance BuildMail ConfirmationMail where
...
 
-TODO
+Every email should have a plain text version for people with reasonable mail clients. If you don't specify one and only set the HTML content via `html` (see above), then IHP automatically creates a plain text version from you by stripping away all HTML tags. This is suboptimal.
+
+The better option is to manually provide a useful plain text version of your emails:
+
+```haskell
+-- don't use, emails look off
+
+text ConfirmationMail { .. } = cs [plain|
```
Comment (mpscholten) [inline Guide/mail.markdown:231]: Actually it's best now to use `import NeatInterpolation` for doing this. This package automatically deals with the indentation. It exports a `[trimming||]` quasiquoter. https://hackage.haskell.org/package/neat-interpolation-0.5.1.2/docs/NeatInterpolation.html

It's already used in many places in IHP itself, [e.g. the code generators](https://github.com/digitallyinduced/ihp/blob/master/IHP/IDE/CodeGen/ViewGenerator.hs#L101).

**Review by mpscholten — APPROVED** [review]

File: Guide/mail.markdown (hunk 1 above)
Comment (hendi) [reply Guide/mail.markdown]: Yes of course Haskell has a proper solution for this :facepalm:  

Docs changed accordingly, thanks!

File: Guide/mail.markdown [hunk 2]
```diff
@@ -107,13 +130,14 @@ config = do
     option $ SMTP
         { host = "smtp.myisp.com"
         , port = 2525
-        , credentials = Nothing -- or Just ("myusername","hunter2")
+        , credentials = Nothing -- or: Just ("myusername","hunter2")
+		, encryption = TLS -- <-- other options: `Unencrypted` or `STARTTLS`
```
Comment (amitaibu) [inline Guide/mail.markdown:134]: Something is off with the indentation here and below

Comment (hendi) [reply Guide/mail.markdown]: Opps, sorry, missed some leftover tabs in there. Thanks for letting me know!

**Review by mpscholten — APPROVED** [review]

**Comment by mpscholten** [comment]
Thanks! 👍

**Pull request merged**